addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/discovery/RioSystemDeviceDiscoveryService.java

```diff
@@ -0,0 +1,270 @@
+     * {@link #thingDiscovered(DiscoveryResult)} for the source.
+     */
+    private void discoverSources() {
+        for (int s = 1; s < 9; s++) {
+            try {
+                final String type = sendAndGet("GET S[" + s + "].type", RSP_SRCNOTIFICATION, 3);
+                if (StringUtils.isNotEmpty(type)) {
+                    final String name = sendAndGet("GET S[" + s + "].name", RSP_SRCNOTIFICATION, 3);
+                    logger.debug("Source #{} - {}/{}", s, type, name);
+
+                    final ThingUID thingUID = new ThingUID(RioConstants.THING_TYPE_SOURCE,
+                            sysHandler.getThing().getUID(), String.valueOf(s));
+
+                    final DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                            .withProperty(RioSourceConfig.Source, s).withBridge(sysHandler.getThing().getUID())
+                            .withLabel((StringUtils.isEmpty(name) || name.equalsIgnoreCase("null") ? "Source" : name)
+                                    + " (" + s + ")")
+                            .build();
+                    thingDiscovered(discoveryResult);
+                }
+            } catch (Exception e) {
+                logger.debug("Exception occurred scanning source #{} : {}", s, e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Helper method to discover zones. This will iterate through all possible zones (8 of them) and see if they
+     * respond to the "name" command. If they do, initial a {@link #thingDiscovered(DiscoveryResult)} for the zone.
+     *
+     * @param controllerUID the {@link ThingUID} of the parent controller
+     * @param c the controller identifier
+     * @throws IllegalArgumentException if controllerUID is null
+     * @throws IllegalArgumentException if c is < 1 or > 8
+     */
+    private void discoverZones(ThingUID controllerUID, int c) {
+        if (controllerUID == null) {
+            throw new IllegalArgumentException("controllerUID cannot be null");
+        }
+        if (c < 1 || c > 8) {
+            throw new IllegalArgumentException("c must be between 1 and 8");
+        }
+        for (int z = 1; z < 9; z++) {
+            try {
+                final String name = sendAndGet("GET C[" + c + "].Z[" + z + "].name", RSP_ZONENOTIFICATION, 4);
+                if (StringUtils.isNotEmpty(name)) {
+                    logger.debug("Controller #{}, Zone #{} found - {}", c, z, name);
+
+                    final ThingUID thingUID = new ThingUID(RioConstants.THING_TYPE_ZONE, controllerUID,
+                            String.valueOf(z));
+
+                    final DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID)
+                            .withProperty(RioZoneConfig.Zone, z).withBridge(controllerUID)
+                            .withLabel((StringUtils.isEmpty(name) || name.equalsIgnoreCase("null") ? "Zone" : name)
```

Comment:
You just checked name `StringUtils.isNotEmpty(name)` on line 218. And can it really equals the string "null", then this should suffice: `name.equalsIgnoreCase("null") ? "Zone" : name`